File: poo_aula2/java/CofreDigital.java
public class CofreDigital{
    
    // Não tem Visibilidade Declarada
    // O Main consegue Visualizar / Editar
    // private -> Outras Classes (como o Main) não podem acessá-la
    private int valor = 0; // Variável Declarado no Corpo da Classe --> Vira um Atributo !!

    // Regra!
    // Método --> public
    // Atributo --> private

    // Método de Depósito
    public void Depositar(int v){
        if(v > 0){
            valor = valor + v;
        }
    }

    // Método para Visualizar o Saldo
    public int Saldo(){
        return valor;
    }

    public boolean Sacar(int v){
        
        if(v <= valor){
            valor = valor - v;
            return true;
        }
        return false;
    }
}

File: poo_aula2/typescript/CofreDigital.ts
export default class CofreDigital{
    
    // Atributo Privado
    private valor: number = 0;

    // Método de Depósito
    Depositar(v: number): void{
        if(v > 0){
            this.valor = this.valor + v;
        }
    }

    // Método para Visualizar Saldo
    Saldo(): number{
        return this.valor;
    }

    // Método de Saque
    Sacar(v: number): boolean{
        if (v <= this.valor){
            this.valor = this.valor - v;
            return true;
        }
        return false; 
    }
}
